metainfo/metainfo.go

```diff
@@ -8,12 +8,24 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 	"time"
 
 	"github.com/anacrolix/torrent/bencode"
 )
 
+type FilesSlice []FileInfo
+
+func (p FilesSlice) Len() int { return len(p) }
+func (p FilesSlice) Less(i, j int) bool {
+	pi := strings.Join(p[i].Path, string(filepath.Separator))
+	pj := strings.Join(p[j].Path, string(filepath.Separator))
```

Comment:
I found sort algorithm in transmissionbt. And to make same files produce identical metainfo I add sort.

https://github.com/wereHamster/transmission/blob/master/libtransmission/makemeta.c

qsort -> builderFileCompare -> asc